tests/manualGrading.test.js

```diff
@@ -0,0 +1,408 @@
+
+/**
+ * @param {object} student or instructor user to load page by
+ * @returns string Returns "Homework for Internal, External, Manual grading methods" page text
+ */
+const loadHomeworkPage = async (user) => {
+  setUser(user);
+  const studentCourseInstanceUrl = baseUrl + '/course_instance/1';
+  const courseInstanceBody = await (await fetch(studentCourseInstanceUrl)).text();
+  const $courseInstancePage = cheerio.load(courseInstanceBody);
+  const hm9InternalExternalManualUrl =
+    siteUrl + $courseInstancePage(`a:contains("${assessmentTitle}")`).attr('href');
+  let res = await fetch(hm9InternalExternalManualUrl);
+  assert.equal(res.ok, true);
+  return res.text();
+};
+
+const loadHomeworkQuestionUrl = async (user) => {
+  const hm1Body = await loadHomeworkPage(user);
+  const $hm1Body = cheerio.load(hm1Body);
+  return siteUrl + $hm1Body(`a:contains("${manualGradingQuestionTitle}")`).attr('href');
+};
+
+/**
+ * Gets the score text for the first submission panel on the page.
+ *
+ * @param {import('cheerio')} $
+ * @returns {string}
+ */
+function getLatestSubmissionStatus($) {
+  return $('.card[id^="submission"] .card-header .badge').first().text();
+}
+
+describe('Manual Grading', function () {
+  this.timeout(80000);
+
+  let iqUrl, iqId;
+  let instancesAssessmentUrl;
+  let manualGradingAssessmentUrl;
+  let manualGradingAssessmentQuestionUrl;
+  let manualGradingIQUrl;
+  let manualGradingNextUngradedUrl;
+  let $manualGradingPage;
+  let score_percent, score_points;
+  let feedback_note;
+
+  before('set up testing server', helperServer.before());
+  after('shut down testing server', helperServer.after);
+
+  before('build assessment manual grading page URL', async () => {
+    const assessments = (await sqlDb.queryAsync(sql.get_assessment, {})).rows;
+    assert.lengthOf(assessments, 1);
+    manualGradingAssessmentUrl = `${baseUrl}/course_instance/1/instructor/assessment/${assessments[0].id}/manual_grading`;
+    instancesAssessmentUrl = `${baseUrl}/course_instance/1/instructor/assessment/${assessments[0].id}/instances`;
+  });
+
+  before('add staff users', async () => {
+    await Promise.all(
+      mockStaff.map(async (staff) => {
+        const courseStaffParams = [1, staff.authUid, 'None', 1];
+        const courseStaffResult = await sqlDb.callAsync(
+          'course_permissions_insert_by_user_uid',
+          courseStaffParams
+        );
+        assert.equal(courseStaffResult.rowCount, 1);
+        staff.user_id = courseStaffResult.rows[0].user_id;
+        const ciStaffParams = [1, staff.user_id, 1, 'Student Data Editor', 1];
+        const ciStaffResult = await sqlDb.callAsync(
+          'course_instance_permissions_insert',
+          ciStaffParams
+        );
+        assert.equal(ciStaffResult.rowCount, 1);
+      })
+    );
+  });
+
+  after('reset default user', () => setUser(defaultUser));
+
+  describe('Submit and grade a manually graded question', () => {
+    it('load page as student', async () => {
```

Comment:
I'd recommend using [`mocha-steps`](https://www.npmjs.com/package/mocha-steps) here for these tests that request specific things to be run sequentially in order. There's prior art in PL, e.g. [here in the active access restriction tests](https://github.com/PrairieLearn/PrairieLearn/blob/0cfb2c71da637f7f4b5b82aebe4e8eaf0ae3cd3d/tests/activeAccessRestriction.test.js#L52).

Alternatively, this could all go into one big `test()`/`it()` function.